src/utils/transition-manager.js

```diff
@@ -0,0 +1,209 @@
+/* global setInterval, clearInterval */
+import assert from 'assert';
+import {
+  viewportLinearInterpolator,
+  extractViewportFrom,
+  VIEWPORT_PROPS
+} from './viewport-transition-utils';
+import MapState from './map-state';
+
+const VIEWPORT_TRANSITION_FREQUENCY = 60; // 60 times per second.
+const VIEWPORT_TRANSITION_EASING_FUNC = t => t;
+
+export const TRANSITION_EVENTS = {
+  BREAK: 1,
+  SNAP_TO_END: 2,
+  IGNORE: 3
+};
+
+const DEFAULT_PROPS = {
+  transitionDuration: 0,
+  transitionInterpolator: viewportLinearInterpolator,
+  transitionEasing: VIEWPORT_TRANSITION_EASING_FUNC,
+  transitionInterruption: TRANSITION_EVENTS.BREAK,
+  onTransitionStart: () => {},
+  onTransitionInterrupt: () => {},
+  onTransitionEnd: () => {}
+};
+
+export default class TransitionManager {
+  constructor(props) {
```

Comment:
I removed the `onTransitionUpdate` callback and have it use `props.onViewportChange`.

Reply:
The reason I initially added `onTransitionUpdate` was to not depend on `onViewportChange` callback to perform transition. Imagine a case if app doesn't provide `onVieportChange` or provides but it doesn't change component's state hence it doesn't trigger the render().

This will binds functionality of transition with application correctly implementing `onViewportChange()`, transition don't occur otherwise.